bigchaindb_driver/pool.py

```diff
@@ -73,7 +104,9 @@ def get_connection(self):
             A :class:`~bigchaindb_driver.connection.Connection` instance.
 
         """
-        if len(self.connections) > 1:
-            return self.picker.pick(self.connections)
 
-        return self.connections[0]
+        if self.tries >= self.max_tries:
```

Comment:
`tries` is incremented every time there is a failure and when it reaches the limit, but never decremented, so if someone is using the driver for enough time to accumulate enough errors, `get_connection` will always return `None`.

In general, I think it's difficult to use a counter to decide if the network is unresponsive or not. Let's say you have 4 nodes, and one of them is always failing (but the other 3 are fine). After some requests, `tries` will reach `max_tries` and `get_connection` will return you `None`, even if the network is responsive.

An idea would be to count how many nodes have been `DELAY`ed, and if all nodes were unavailable, then raise an exception and reset the nodes `DELAY`s. What do you think?

Reply:
I Agree. so i updated the implementation, using:

1. A new dict ```retries_left``` per node, to keep track of retries left per node. 
2. Updating ```retries_left``` based not only when node is failing but also when node is successful. 
3. and Return ```None # Service Unavailable``` when  ```retries_left per node == 0``` for all nodes.

Reply:
Couple of things to clarify item 2 and 3.
> 2. Updating retries_left based not only when node is failing but also when node is successful.

```retries_left``` per node is based on max and min constraints 
```
min_retries = 0
max_retries = len(nodes)*4. #could be another max limit
```

> 3. and Return None # Service Unavailable when retries_left per node == 0 for all nodes.

I think there is a problem particularly if the list of nodes is very big. Lets consider a worst case scenario:
50 nodes: 1 is ok, 49 failing. So the round_robin will try the 49 nodes after reaching the correct one again. That suggests a limit to the number of nodes, what do you think?